fubectl.source

```diff
@@ -14,17 +14,22 @@ alias kwall="watch kubectl get pods --all-namespaces"
 # [kp] open kubernetes dashboard with proxy
 alias kp="open 'http://localhost:8001/ui' && kubectl proxy"
 
+# Return ture if the given resource name is namespaced in k8s
+_is_non_namespaced_resource() {
+  non_namespaced_resource_names=$(kubectl api-resources --namespaced=false --no-headers=true | awk '{ print $1,$2 }' | tr ',' ' ' | tr '\n' ' ')
```

Comment:
`awk` use of `$1` and `$2` does not work as expected here because there is no input field separator. The separation is instead based on column width. So this function will include the APIGROUPS column in the evaluation for lines which have no SHORTNAME.

As an alternative I propose the function:
```
_isNamespaced() {
  obj="$1"
  if [ -z "$obj" ]; then
    echo "Please specify a Kubernetes object!"
    exit 1
  fi  

  kubectl api-resources --namespaced=true \
    | awk '(idx){print substr($0, 0, idx) } (!idx){ idx=index($0, " APIGROUP")}' \
    | grep -qE "\<${obj}s?\>"
}
```

It is using `awk` and the header to split by column. A simple grep is doing a word-boundary split. The plural is handled in the grep pattern with the `s?`.